### PR TITLE
First Attempt at getting the GoogleSheet Api to work

### PR DESCRIPTION
**Pull request by Louis345**
Let me know if you guys have any issues in testing this.

### CODE REVIEW

File: routes/dataVisualization.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            isSuccess: 0,
+            err
+          });
+        }
+      }
+    );
+  }
+});
+
+mapData = tableData => {
```
Comment (queensburybill) [inline routes/dataVisualization.js]: I think you need to declare mapData as a constant here.

File: routes/dataVisualization.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  const header = tableData[0];
+  const results = tableData.map(data => {
+    const obj = data.reduce((acc, item, index) => {
+      acc[header[index]] = item;
```
Comment (queensburybill) [inline routes/dataVisualization.js]: Is it your intention for the first object in the array to be the column headers? Is that what the students are expecting?

**Comment by queensburybill** [comment]
One overall thing I don't understand is how that token works. I had to manually visit a webpage and then paste the token into the command line. How will that work in production?

File: routes/dataVisualization.js (hunk 2 above)
Comment (Louis345) [reply routes/dataVisualization.js]: Yes it is.

File: routes/dataVisualization.js (hunk 1 above)
Comment (Louis345) [reply routes/dataVisualization.js]: Thanks, nice catch.

File: credentials.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "installed": {
+    "client_id": "883194266014-3bo9tunf5br7ihhps1bbmhlslf61b9p9.apps.googleusercontent.com",
+    "project_id": "quickstart-1557075849757",
+    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+    "token_uri": "https://oauth2.googleapis.com/token",
+    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+    "client_secret": "IYW9FNMejPudWmg0rpng5daL",
+    "redirect_uris": ["urn:ietf:wg:oauth:2.0:oob", "http://localhost"]
+  }
```
Comment (matinaspatsos) [inline credentials.json]: This should **NOT** be hardcoded into the a separate file. Please use environment variables and the _config.js_ file instead.

File: app.js [hunk 4]
```diff
@@ -49,9 +54,10 @@ app.use("/api/graduates", graduatesRouter);
 app.use("/api/login", loginRouter);
 app.use("/api/upload/", uploadRouter);
 app.use("/api/download/resumes", downloadResumesRouter);
+app.use("/api/graduates/dataVisualization", dataVisualization);
+
```
Comment (matinaspatsos) [inline app.js]: Route should be _/api/graduates/data-visualization

API URLs

**Pull request merged**